### ISSUE TITLE
Global keyboard shortcuts

### ISSUE THREAD

**Issue by defaude** [overlayeddev/overlayed#201]
**Is your feature request related to a problem? Please describe.**

It would be nice to have global keyboard shortcuts to control Overlayed features like e.g.
- toggling pined / unpinned mode to hide / show the channel tab header w/ buttons
- opening settings window
- quitting Overlayed
- etc.

What do you  think?

Tauri should provide most of the heavy lifting needed to support global shortcuts already, so the implementation _should_ not be too much effort, I think.

**Describe the solution you'd like**

Ideally, the settings UI added a new section for the actions that can be bound, where the user can set up their bindings. This could be done via actually having a "listener" mode where the user can just press the desired shortcut. Regarding the syntax for storing the shortcuts to the config: Why not just use [Tauri's format](https://tauri.app/v1/api/js/globalShortcut#register)?

These shortcuts should be - obviously - global, i.e. I can press the shortcut while I'm in a game, for example.

**Describe alternatives you've considered**

Alternatively (and simpler to implement), we could allow inputting a string that declares the desired shortcut, e.g. something like `CtrlAltShift+o`).

Even simpler is to just add an entry in our `config.json` without any GUI to start with.

**Comment by Hacksore** [comment]
yeah the one challenge is we really need to get the [tauri store plugin](https://github.com/overlayeddev/overlayed/issues/202) added first so we can have access to the config in rust land.

Then for the first phase of this we can allow users to set them in the config, something like how VSCode does it.

```
{
  "hotkeys": [
    {
        "key": "cmd+shift+1",
        "command": "pin"
    },
  ]
}
```
